### PR TITLE
chore!: add error return params to tree interface

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/rsmt2d/issues/156

### CODE REVIEW

File: datasquare.go [hunk 1]
```diff
@@ -198,12 +198,20 @@ func (ds *dataSquare) computeRoots() {
...
+			}
+			rowRoots[i] = rowRoot
 		}(i)
 
 		go func(i uint) {
 			defer wg.Done()
-			colRoots[i] = ds.getColRoot(i)
+			colRoot, err := ds.getColRoot(i)
+			if err != nil {
+				panic(err)
```
Comment (evan-forbes) [inline datasquare.go:212]: I know this probably shouldn't get hit and its in a goroutine, but is there a way to bubble these up as well? computing the roots is

**Review by evan-forbes — COMMENTED** [review]
nice! thanks for doing this, I think its a really good change!

only had a single comment, but otherwise LGTM :+1:

File: datasquare.go (hunk 1 above)
Comment (Wondertan) [reply datasquare.go:212]: errgroup

Comment (rootulp) [reply datasquare.go]: Thanks for the tip @Wondertan ! Resolved in https://github.com/celestiaorg/rsmt2d/pull/157/commits/bbacca88a5448fa16df25f9aeeece7e86c535fff

**Comment by evan-forbes** [comment]
cc @Wondertan as this api change would have an effect on node

**Review by evan-forbes — APPROVED** [review]

File: datasquare.go [hunk 2]
```diff
@@ -187,29 +189,41 @@ func (ds *dataSquare) resetRoots() {
 	}
 }
 
-func (ds *dataSquare) computeRoots() {
-	var wg sync.WaitGroup
+func (ds *dataSquare) computeRoots() error {
+	g := new(errgroup.Group)
```
Comment (Wondertan) [inline datasquare.go:193]: New forces allocation on heap. There is no reason to do it here as it used within methods boundaries and should be kept on stack by var

Comment (rootulp) [reply datasquare.go]: Does https://github.com/celestiaorg/rsmt2d/pull/157/commits/1956b1665b2669f945ecf23ce31d64d1211dd989 resolve your concern?

Comment (Wondertan) [reply datasquare.go]: Yes

Comment (Wondertan) [reply datasquare.go]: The quickest link I found: https://stackoverflow.com/a/10866871

**Review by Wondertan — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

**Comment by rootulp** [comment]
It looks like @Wondertan doesn't have write permissions on this repo so his approval doesn't satisfy the 2 approvals requirement. I removed @MSevey 's review request b/c he's on parental leave so this needs @liamsi 's review.

**Comment by Wondertan** [comment]
And @liamsi is ooo as well, maybe @musalbas ?

**Comment by rootulp** [comment]
Maybe @cmwaters has write permission to this repo b/c he's a member of core/app

**Review by SweeXordious — APPROVED** [review]

**Pull request merged**